src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2687,6 +2687,49 @@ LogicalResult ONNXConstantOp::inferShapes(
 // Concat
 //===----------------------------------------------------------------------===//
 
+static LogicalResult verify(ONNXConcatOp op){
+  ONNXConcatOpAdaptor operandAdaptor =
+    ONNXConcatOpAdaptor(op);
+
+  int inputNum = op.getNumOperands();
+  // Check all inputs.
+  for (int i = 0; i < inputNum; ++i) {
```

Comment:
We try to follow LLVM coding guidelines, those guidelines indicate it is prefered to use for-range rather than a C-like for loop  (see https://llvm.org/docs/CodingStandards.html#use-range-based-for-loops-wherever-possible) although old code might not follow these rules (if you see it feel free to fix it).